www/content/workstation/upgrade_labs.md

```diff
@@ -0,0 +1,454 @@
+
+### Server Backup
+
+We recommend performing a backup before starting any server upgrade process.
+Follow the [Chef Infra Server Backup](https://docs.chef.io/runbook/server_backup_restore/) documentation before starting your Upgrade Lab.
+
+### Chef Infrastructure Requirements
+
+#### Credentials
+
+Your credentials are set up using [knife profiles](https://docs.chef.io/workstation/knife_setup/#knife-profiles). This allows you to keep your keys in a `credentials` file, and makes switching between credentials easier.
+
+For example, in `.chef/credentials`:
+
+```toml
+[old-server]
+client_name = "myuser"
+chef_server_url = "https://old-chef-server.dev/organizations/my-org"
+client_key = """
+-----BEGIN RSA PRIVATE KEY-----
+MMM+some+key+goes+here+MMM
+-----END RSA PRIVATE KEY-----
+"""
+[new-server]
+client_name = "myuser"
+chef_server_url = "https://new-chef-server.dev/organizations/my-org"
+client_key = """
+-----BEGIN RSA PRIVATE KEY-----
+MMM+another+key+goes+here+MMM
+-----END RSA PRIVATE KEY-----
+"""
+```
+
+#### Connectivity
+
+* You have a user key for both of the Chef Infra Servers
+* You can connect to both Chef Infra Servers from your development workstation.
+
+Verify connectivity by running a knife command against each server and receive a reasonable response:
+
+```shell
+chef exec knife user list --profile old-server my-user
+chef exec knife user list --profile new-server my-user
+```
+
+#### Convergence
+
+Your nodes are in a good working order. They converge cleanly under the older version of Chef Infra Client.
+
+Verify that the nodes are healthy by running:
+
+```shell
+chef exec knife status --profile old-server
+```
+
+Which outputs something similar to:
+
+```output
+42 minutes ago, node-01, ubuntu 18.04.
+```
+
+This command outputs the time of the last successful chef run of each node; nodes that return radically different times for the last successful chef run are not in good working order.
+
+#### Cookbook CI/CD
+
+While we don't prescribe a particular choice of technology or the details of processes, the Chef Upgrade Lab expects a continuous integration pipeline and continuous delivery system (CI/CD) for cookbook deployments.
+
+The Upgrade Lab assumes--but does not require--that you have a continuous integration pipeline (CI) setup for your cookbooks with:
+
+  * A version control system (for example, git)
+  * Some degree of automated testing for proposed changes
+  * A continuous delivery system (CD) that controls cookbook releases; the CD is the mechanism for updating cookbook versions and uploading them to the Chef Server(s)
+
+If you do not have a version control system and CI/CD pipeline in place, then please contact [Chef Software customer support](https://www.chef.io/support/).
+
+#### Cookbook Locations
```

Comment:
I think this section should come later. We're telling people to go searching for cookbooks, but they don't know which cookbooks to search for, because they have not run the report yet. In fact, `capture` interactively tells you exactly which cookbooks it needs to locate.